skopt/plots.py

```diff
@@ -93,322 +98,880 @@ def plot_convergence(*args, **kwargs):
+    bounds2 = dimension2.bounds
+
+    # Generate evenly spaced points between the dimension bounds.
+    xi = np.linspace(bounds1[0], bounds1[1], n_points)
+    yi = np.linspace(bounds2[0], bounds2[1], n_points)
+
+    # Transform the points if necessary.
+    xi_transformed = dimension1.transform(xi)
+    yi_transformed = dimension2.transform(yi)
 
-        yi = []
-        for x_ in xi_transformed:
-            rvs_ = np.array(sample_points)
-            rvs_[:, i] = x_
-            yi.append(np.mean(model.predict(rvs_)))
+    # Calculate the partial dependence for all combinations of these points.
+    zi = [[_calc(x, y) for x in xi_transformed] for y in yi_transformed]
 
-        return xi, yi
+    # Convert list-of-list to a numpy array.
+    zi = np.array(zi)
 
+    return xi, yi, zi
+
+
+def plot_evaluations(result, bins=20, dimension_names=None):
+    """
+    Visualize the order in which points were sampled during optimization.
+
+    This creates a 2-d matrix plot where the diagonal plots are histograms
+    that show the distribution of samples for each search-space dimension.
+
+    The plots below the diagonal are scatter-plots of the samples for
+    all combinations of search-space dimensions.
+
+    The ordering of the samples are shown as different colour-shades.
+
+    A red star shows the best found parameters.
+
+    NOTE: Search-spaces with `Categorical` dimensions are not supported.
+
+    Parameters
+    ----------
+    * `result` [`OptimizeResult`]
+        The optimization results from calling e.g. `gp_minimize()`.
+
+    * `bins` [int, bins=20]:
+        Number of bins to use for histograms on the diagonal.
+
+    * `dimension_names` [list(str)]:
+        List of names for search-space dimensions to be used in the plot.
+        You can omit `Categorical` dimensions here as they are not supported. 
+        If `None` then use all dimensions from the search-space.
+
+    Returns
+    -------
+    * `fig`: [`Matplotlib.Figure`]:
+        The object for the figure.
+        For example, call `fig.savefig('plot.png')` to save the plot.
+
+    * `ax`: [`Matplotlib.Axes`]:
+        A 2-d matrix of Axes-objects with the sub-plots.
+    """
+
+    # Get the search-space instance from the optimization results.
+    space = result.space
+
+    # Get the relevant search-space dimensions.
+    if dimension_names is None:
+        # Get all dimensions.
+        dimensions = space.dimensions
     else:
-        # XXX use linspace(*bounds, n_points) after python2 support ends
-        bounds = space.dimensions[j].bounds
-        xi = np.linspace(bounds[0], bounds[1], n_points)
-        xi_transformed = space.dimensions[j].transform(xi)
+        # Only get the named dimensions.
+        dimensions = space[dimension_names]
+
+    # Ensure there are no categorical dimensions.
+    # TODO replace with check_list_types(dimensions, (Integer, Real)) in PR #597
+    if any(type(dim) == Categorical for dim in dimensions):
```

Comment:
use `isinstance` to check the type of an object in python

Reply:
Right. However, this should be replaced with check_list_types from PR #597 as mentioned in the TODO-comment. I am waiting for a review of that PR.

Reply:
Fixed. The correct thing would be to use PR #597 but it has not yet been merged and I want this PR to work as-is.